modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/SpringBundleResourceContributor.java

```diff
@@ -0,0 +1,94 @@
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.capability.xml;
+
+import org.mule.extension.introspection.Extension;
+import org.mule.extension.introspection.capability.XmlCapability;
+import org.mule.extension.resources.ResourcesGenerator;
+import org.mule.extension.resources.spi.GenerableResourceContributor;
+import org.mule.module.extension.internal.capability.xml.schema.SchemaGenerator;
+import org.mule.module.extension.internal.capability.xml.schema.model.SchemaConstants;
+import org.mule.module.extension.internal.config.ExtensionsNamespaceHandler;
+
+import java.util.Collection;
+
+import org.apache.commons.collections.CollectionUtils;
+
+/**
+ * Implementation of {@link org.mule.extension.resources.spi.GenerableResourceContributor}
+ * spring bundles
+ *
+ * @since 3.7.0
+ */
+public class SpringBundleResourceContributor implements GenerableResourceContributor
+{
+
+    @Override
+    public void contribute(Extension extension, ResourcesGenerator resourcesGenerator)
+    {
+        Collection<XmlCapability> capabilities = extension.getCapabilities(XmlCapability.class);
+        if (CollectionUtils.isEmpty(capabilities))
+        {
+            return;
+        }
+
+        XmlCapability capability = capabilities.iterator().next();
+
+        generateSchema(extension, capability, resourcesGenerator);
+        generateSpringBundle(extension, capability, resourcesGenerator);
+    }
+
+    private void generateSchema(Extension extension, XmlCapability capability, ResourcesGenerator resourcesGenerator)
+    {
+        String schema = new SchemaGenerator().generate(extension, capability);
+        resourcesGenerator.getOrCreateResource(getXsdFileName(extension)).getContentBuilder().append(schema);
+    }
+
+    private void generateSpringBundle(Extension extension, XmlCapability capability, ResourcesGenerator resourcesGenerator)
+    {
+        writeSpringHandlerBundle(capability, resourcesGenerator);
+        writeSpringSchemaBundle(extension, capability, resourcesGenerator);
+    }
+
+    private void writeSpringHandlerBundle(XmlCapability capability, ResourcesGenerator resourcesGenerator)
+    {
+        String content = String.format("%s=%s", capability.getSchemaLocation(), ExtensionsNamespaceHandler.class.getName());
+        resourcesGenerator.getOrCreateResource("spring.handlers").getContentBuilder().append(springBundleScape(content));
+    }
+
+    private void writeSpringSchemaBundle(Extension extension, XmlCapability capability, ResourcesGenerator resourcesGenerator)
+    {
+        StringBuilder builder = resourcesGenerator.getOrCreateResource("spring.schemas").getContentBuilder();
+        builder.append(getSpringSchemaBundle(extension, capability, capability.getSchemaVersion()));
+        builder.append(getSpringSchemaBundle(extension, capability, "current"));
```

Comment:
Current mule spring.schemas files contain definitions for multiple mule versions. Need to review how that is going to be provided with the extensions